File: Python/61-70/e_65.py
# Escribe una función en Python numOcurrencias(L), que tome como parámetro una lista L y devuelva una lista de tuplas,
# donde cada tupla corresponda a un elemento de la lista L con su número de ocurrencias en la lista.


def numOcurrencias(L):
    ocurrencias = {}
    for elemento in L:
        if elemento in ocurrencias:
            ocurrencias[elemento] += 1
        else:
            ocurrencias[elemento] = 1
    return [(elemento, ocurrencias[elemento]) for elemento in ocurrencias]

# Ejemplo de uso:
lista = [1, 2, 3, 4, 2, 5, 2]
resultado = numOcurrencias(lista)
print(resultado)


# Esta función crea un diccionario ocurrencias donde las claves son los elementos únicos de la lista y los valores son el número de ocurrencias de cada elemento. 
# Luego, crea una lista de tuplas donde cada tupla contiene un elemento de la lista junto con su número de ocurrencias,
# utilizando una comprensión de listas. Finalmente, devuelve esta lista de tuplas.

File: JS/61-70/e_65.js
function numOcurrencias(L) {
    var ocurrencias = {};
    for (var i = 0; i < L.length; i++) {
        var elemento = L[i];
        if (ocurrencias[elemento]) {
            ocurrencias[elemento] += 1;
        } else {
            ocurrencias[elemento] = 1;
        }
    }
    var resultado = [];
    for (var clave in ocurrencias) {
        resultado.push([parseInt(clave), ocurrencias[clave]]);
    }
    return resultado;
}

// Ejemplo de uso:
var lista = [1, 2, 3, 4, 2, 5, 2];
var resultado = numOcurrencias(lista);
console.log(resultado);


// Esta función crea un objeto ocurrencias,
// donde las claves son los elementos únicos de la lista y los valores son el número de ocurrencias de cada elemento.
// Luego, recorre este objeto y construye una matriz resultado donde cada elemento es una matriz con dos valores: 
// el elemento y su número de ocurrencias. Finalmente, devuelve esta matriz resultado.




